### PR TITLE
Feature/ti2023 36/backend browse events

### PR DESCRIPTION
**Pull request by Filip-B-32**

### CODE REVIEW

File: EventPlannerAPI/BusinessLayer/DTOs/PaginationFilter.cs [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        public double MaximumDuration { get; set; }
+        public string Location { get; set; }
+        public string AuthorUserId { get; set; }
+        public int SkillLevel { get; set; }
+
+        public PaginationFilter()
+        {
+            this.PageNumber = 1;
+            this.PageSize = 5;
+            this.SearchData = string.Empty;
```
Comment (HellHound97) [inline EventPlannerAPI/BusinessLayer/DTOs/PaginationFilter.cs]: why populate with string empty?

File: EventPlannerAPI/BusinessLayer/DTOs/PaginationFilter.cs [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+            this.SportTypeId= Guid.Empty;
+            this.StartDate = new DateTime();
+            this.MaximumDuration = 0.0;
+            this.Location = string.Empty;
```
Comment (HellHound97) [inline EventPlannerAPI/BusinessLayer/DTOs/PaginationFilter.cs]: why populate with string empty?

File: EventPlannerAPI/BusinessLayer/DTOs/PaginationFilter.cs [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+            this.AuthorUserId = string.Empty;
```
Comment (HellHound97) [inline EventPlannerAPI/BusinessLayer/DTOs/PaginationFilter.cs]: why populate with string empty?

File: EventPlannerAPI/BusinessLayer/DTOs/PaginationFilter.cs [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+        public int SkillLevel { get; set; }
+
+        public PaginationFilter()
+        {
+            this.PageNumber = 1;
+            this.PageSize = 5;
+            this.SearchData = string.Empty;
+            this.SportTypeId= Guid.Empty;
+            this.StartDate = new DateTime();
+            this.MaximumDuration = 0.0;
```
Comment (HellHound97) [inline EventPlannerAPI/BusinessLayer/DTOs/PaginationFilter.cs]: Why 0.0?

File: EventPlannerAPI/BusinessLayer/DTOs/PaginationFilter.cs [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+            this.SearchData = searchData;
+            this.SportTypeId = sportTypeId;
+            this.StartDate = startDate;
+            this.MaximumDuration = maximumDuration;
+            this.Location = location;
+            this.AuthorUserId = authorUserId;
+            this.SkillLevel = skillLevel;
+        }
+
+        public bool IsEmpty()
```
Comment (HellHound97) [inline EventPlannerAPI/BusinessLayer/DTOs/PaginationFilter.cs]: why?

File: EventPlannerAPI/BusinessLayer/Interfaces/IEventService.cs [hunk 6]
```diff
@@ -1,5 +1,6 @@
 ﻿using BusinessLayer.DTOs;
 using DataAccessLayer.Models;
+using Microsoft.AspNetCore.Mvc;
```
Comment (HellHound97) [inline EventPlannerAPI/BusinessLayer/Interfaces/IEventService.cs]: why?

File: EventPlannerAPI/DataAccessLayer/Repositories/EventRepository.cs [hunk 7]
```diff
@@ -3,6 +3,7 @@
 using DataAccessLayer.Interfaces;
 using DataAccessLayer.Models;
 using Microsoft.EntityFrameworkCore;
+using System.Drawing.Printing;
```
Comment (HellHound97) [inline EventPlannerAPI/DataAccessLayer/Repositories/EventRepository.cs]: what is this?

File: EventPlannerAPI/DataAccessLayer/Repositories/EventRepository.cs [hunk 8]
```diff
@@ -39,14 +40,25 @@ public async Task<Event> GetEventByIdAsync(Guid eventId)
...
+                    ((maximumDuration <= 0) || (evnt.EndDate - evnt.StartDate).TotalHours <= maximumDuration) &&
+                    (string.IsNullOrEmpty(location) || evnt.Location.Contains(location)) &&
+                    (string.IsNullOrEmpty(authorUserId) || evnt.Author.Id == authorUserId) &&
+                    evnt.SkillLevel == skillLevel) 
+                .Skip((pageNumber - 1) * pageSize)
+                .Take(pageSize)
                 .ToListAsync();
         }
 
+
```
Comment (HellHound97) [inline EventPlannerAPI/DataAccessLayer/Repositories/EventRepository.cs:60]: remove

**Review by HellHound97 — APPROVED** [review]

**Review by alexandratodea — APPROVED** [review]

**Pull request merged**